<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>765</width>
    <height>437</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Neural Network</string>
  </property>
  <property name="windowIcon">
   <iconset resource="res.qrc">
    <normaloff>:/icon.ico</normaloff>:/icon.ico</iconset>
  </property>
  <property name="windowOpacity">
   <double>0.990000000000000</double>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="iconSize">
   <size>
    <width>512</width>
    <height>512</height>
   </size>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QGroupBox" name="NNinputs">
        <property name="locale">
         <locale language="English" country="UnitedStates"/>
        </property>
        <property name="title">
         <string/>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QLabel" name="trainDataLabel">
             <property name="locale">
              <locale language="English" country="UnitedStates"/>
             </property>
             <property name="text">
              <string>Training data file:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="hiddenNeuronLabel">
             <property name="locale">
              <locale language="English" country="UnitedStates"/>
             </property>
             <property name="text">
              <string>Hidden layer neurons:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLineEdit" name="trainDataPathEdit">
               <property name="minimumSize">
                <size>
                 <width>275</width>
                 <height>0</height>
                </size>
               </property>
               <property name="locale">
                <locale language="English" country="UnitedStates"/>
               </property>
               <property name="inputMask">
                <string/>
               </property>
               <property name="dragEnabled">
                <bool>false</bool>
               </property>
               <property name="placeholderText">
                <string>Enter file path or click button</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="trainDataPathButton">
               <property name="locale">
                <locale language="English" country="UnitedStates"/>
               </property>
               <property name="text">
                <string>C:\</string>
               </property>
               <property name="icon">
                <iconset theme="document-open"/>
               </property>
               <property name="shortcut">
                <string>Ctrl+O</string>
               </property>
               <property name="popupMode">
                <enum>QToolButton::DelayedPopup</enum>
               </property>
               <property name="toolButtonStyle">
                <enum>Qt::ToolButtonIconOnly</enum>
               </property>
               <property name="autoRaise">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QSpinBox" name="hiddenNeuronCount">
               <property name="locale">
                <locale language="English" country="UnitedStates"/>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>1000</number>
               </property>
               <property name="value">
                <number>1</number>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <item>
                <widget class="QLabel" name="epochLabel">
                 <property name="locale">
                  <locale language="English" country="UnitedStates"/>
                 </property>
                 <property name="text">
                  <string>Epochs:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="epochInput">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="locale">
                  <locale language="English" country="UnitedStates"/>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>1000000</number>
                 </property>
                 <property name="value">
                  <number>10000</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="NNactions">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="locale">
         <locale language="English" country="UnitedStates"/>
        </property>
        <property name="title">
         <string/>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QPushButton" name="initButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string>Initialize neural network</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="trainButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="locale">
            <locale language="English" country="UnitedStates"/>
           </property>
           <property name="text">
            <string>Train neural network</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="progressBars">
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QProgressBar" name="trainProcess">
         <property name="styleSheet">
          <string notr="true">QProgressBar {
    border: 1px solid #252525;
    border-radius: 10px;
    background-color: #2d2d2d;
    text-align: center;
    color: #ffffff;
}

QProgressBar::chunk {
    background: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, 
                                stop:0 rgba(252,151,155,255), 
                                stop:0.5 rgba(242,111,115,255), 
                                stop:1 rgba(252,151,155,255));
    border-radius: 5px;
}
</string>
         </property>
         <property name="locale">
          <locale language="English" country="UnitedStates"/>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="format">
          <string>Training process %p%</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="IOfields">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <property name="title">
       <string>Input / output fields</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QListWidget" name="inputList">
         <property name="font">
          <font>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="layoutDirection">
          <enum>Qt::RightToLeft</enum>
         </property>
         <property name="styleSheet">
          <string notr="true">QListWidget {
    border: 1px solid #252525;
    border-radius: 2px;
}</string>
         </property>
         <property name="locale">
          <locale language="English" country="UnitedStates"/>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::AllEditTriggers</set>
         </property>
         <property name="tabKeyNavigation">
          <bool>true</bool>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::NoSelection</enum>
         </property>
         <property name="movement">
          <enum>QListView::Static</enum>
         </property>
         <property name="flow">
          <enum>QListView::TopToBottom</enum>
         </property>
         <property name="uniformItemSizes">
          <bool>false</bool>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="selectionRectVisible">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="outputList">
         <property name="font">
          <font>
           <pointsize>14</pointsize>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">QListWidget {
    border: 1px solid #252525;
    border-radius: 2px;
}</string>
         </property>
         <property name="locale">
          <locale language="English" country="UnitedStates"/>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::NoSelection</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections/>
</ui>
